File: Examples/Features/PDFExporting/ExportShape/CustomShape.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.PDFExporting.ExportShape
{
    public class CustomShape : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            //Open an excel file
            var fileStream = this.GetResourceStream("xlsx\\CustomShapes.xlsx");
            workbook.Open(fileStream);
        }
        
        public override bool SavePdf
        {
            get
            {
                return true;
            }
        }

        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }
        public override string[] UsedResources
        {
            get
            {
                return new string[] { "xlsx\\CustomShapes.xlsx" };
            }
        }
    }
}

File: Examples.VB/Features/PDFExporting/ExportShape/CustomShape.vb
Namespace Features.PDFExporting.ExportShape
    Public Class CustomShape
        Inherits ExampleBase

        Public Overrides Sub Execute(workbook As Excel.Workbook)
            'Open an excel file
            Dim fileStream = GetResourceStream("CustomShapes.xlsx")
            workbook.Open(fileStream)
        End Sub

        Public Overrides ReadOnly Property SavePdf As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property ShowScreenshot As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides ReadOnly Property UsedResources As String()
            Get
                Return New String() {"xlsx\CustomShapes.xlsx"}
            End Get
        End Property
    End Class
End Namespace
